### DISCUSSION TITLE
Real-time WPF plotting optimization

### DISCUSSION THREAD

**Discussion by alxkalinin** [oxyplot/oxyplot#1841] [Q&A]
I use OxyPlot WPF for real-time plotting. And it works quite good, FPS=60 up to number of points 100000. But I did a quick profiling and found an interesting thing. A lot of the time OxyPlot spends in "AdjustPlotMargins" method

![image](https://user-images.githubusercontent.com/97022983/150812063-63615777-9a60-47ed-98d5-38913bd21a67.png)

But from frame to frame my Axis still the same. I move only points, the axis do not change. Is there any way to say OxyPlot do not make new measure, because all measures still the same? It seems it improve performance a lot.

**Comment by alxkalinin** [comment]
My own solution is to overlay two plots. One is static for axis drawing and the other for real-time plot update, no axis and only plot update. With this approach I have significant improvement in real-time plotting performance.

So, it works. But may be we can do it without overlaying.

**Reply by professordave** [reply]
Interesting. You say it is real time and the axis does not change. 

Is time plotting against the x-axis? 

Is it like a strip chart, then the x-axis would be moving to the right?

I am plotting some values against time, suppose new points are always within the existing y range, then the issue is the x-axis.

**Reply by alxkalinin** [reply]
The general idea is
- the **axis redrawing** process in real-time takes a sufficient amount of time
- we can reduce the axis redrawing time by overlaying two plots, one for the axis and the other for  plot updates

And here, it could be different strategies for updating the plot and axis. My strategy is to use a no-scrolling plot update, like in typical [ECG monitors](https://youtu.be/X3HgZqw_3ao). In this strategy, I can update axis one per window change. It gives us much better performance.
